layers/dns.go

```diff
@@ -464,16 +465,32 @@ func (rr *DNSResourceRecord) String() string {
 	return "..."
 }
 
+func decodeCharacterStrings(data []byte) ([][]byte, error) {
+	var strings [][]byte
```

Comment:
not sure I like the 3-4 mallocs this could cause as the slice grows... maybe start with:

strings := make([][]byte, 0, 4)

so that for most cases we only get one?  None would be better, but what can you do?